src/charm.py

```diff
@@ -78,6 +110,171 @@ def configure_magma_access_gateway(orc8r_domain: str, root_ca_path: str) -> None
+        ipv4_gateway = self.model.config.get(f"{interface}-ipv4-gateway")
+        ipv6_address = self.model.config.get(f"{interface}-ipv6-address")
+        ipv6_gateway = self.model.config.get(f"{interface}-ipv6-gateway")
+        if not ipv4_address and not ipv4_gateway and not ipv6_address and not ipv6_gateway:
+            return True
+        if any([ipv4_address, ipv4_gateway]) and not all([ipv4_address, ipv4_gateway]):
+            logger.warning("Both IPv4 address and gateway required for interface %s", (interface))
+            return False
+        if any([ipv6_address, ipv6_gateway]) and not all([ipv6_address, ipv6_gateway]):
+            logger.warning("Both IPv6 address and gateway required for interface %s", (interface))
+            return False
+        if ipv6_address and not ipv4_address:
+            logger.warning(
+                "Pure IPv6 configuration is not supported for interface %s", (interface)
+            )
+            return False
+        if ipv4_address and not self._is_valid_ipv4_address(ipv4_address):
+            logger.warning("Invalid IPv4 address and netmask for interface %s", (interface))
+            return False
+        if ipv4_gateway and not self._is_valid_ipv4_gateway(ipv4_gateway):
+            logger.warning("Invalid IPv4 gateway for interface %s", (interface))
+            return False
+        if ipv6_address and not self._is_valid_ipv6_address(ipv6_address):
+            logger.warning("Invalid IPv6 address and netmask for interface %s", (interface))
+            return False
+        if ipv6_gateway and not self._is_valid_ipv6_gateway(ipv6_gateway):
+            logger.warning("Invalid IPv6 gateway for interface %s", (interface))
+            return False
+        return True
+
+    def _is_valid_ipv4_address(self, ipv4_address: str) -> bool:
+        """Validate an IPv4 address and netmask.
+
+        A valid string will have the form:
+        a.b.c.d/x
+        """
+        try:
+            ip = ipaddress.ip_network(ipv4_address, strict=False)
+            return isinstance(ip, ipaddress.IPv4Network)
+        except ValueError:
+            return False
+
+    def _is_valid_ipv4_gateway(self, ipv4_gateway: str) -> bool:
+        """Validate an IPv4 gateway.
+
+        A valid string will have the form:
+        a.b.c.d
+        """
+        try:
+            ip = ipaddress.ip_address(ipv4_gateway)
+            return isinstance(ip, ipaddress.IPv4Address)
+        except ValueError:
+            return False
+
+    def _is_valid_ipv6_address(self, ipv6_address: str) -> bool:
+        """Validate an IPv6 address and netmask."""
+        try:
+            ip = ipaddress.ip_network(ipv6_address, strict=False)
+            return isinstance(ip, ipaddress.IPv6Network)
+        except ValueError:
+            return False
+
+    def _is_valid_ipv6_gateway(self, ipv6_gateway: str) -> bool:
+        """Validate an IPv6 gateway."""
```

Comment:
Same for the gateway?

Reply:
Fixed, same comment regarding the IPv6 format.